File: CF 894 div 3/c.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int t;
    cin >> t;
    
    for (int _ = 0; _ < t; ++_) {
        int n;
        cin >> n;
        
        vector<int> arr(n);
        for (int i = 0; i < n; ++i) {
            cin >> arr[i];
        }
        
        vector<int> newarr(arr[0], 0);
        for (int ele : arr) {
            newarr[ele - 1]++;
        }
        
        int s = 0;
        bool f = false;
        
        if (arr.size() == static_cast<size_t>(arr[0])) {
            for (int i = newarr.size() - 1; i >= 0; --i) {
                s += newarr[i];
                if (arr[i] != s) {
                    f = true;
                    break;
                }
            }
            if (f) {
                cout << "NO" << endl;
            } else {
                cout << "YES" << endl;
            }
        } else {
            cout << "NO" << endl;
        }
    }
    
    return 0;
}

File: CF 894 div 3/c.py
for _ in range(int(input())):
    n=int(input())
    arr=list(map(int,input().split()))
    if arr[0]>(10**5)*2:
        print('NO')
    else:
        newarr=[0]*arr[0]
        for ele in arr:
            newarr[ele-1]+=1
        s=0
        f=False
        if len(arr)==arr[0]:
            for i in range(len(newarr)-1,-1,-1):
                s+=newarr[i]
                if arr[i]!=s:
                    f=True
                    break
            if f:
                print('NO')
            else:
                print('YES')
        else:
            print('NO')
